### PR TITLE
fix: support text-only email configs

### LINKED ISSUE #5421: Unable to send text email alerts in Rancher monitoring
**What happened?**
In Rancher monitoring(chart version 100.2.0+up40.1.2), unable to sent text email alerts even after doing necessary config. 

**Did you expect to see something different?**
After explicitly providing text template for email alerts, we should get only text alerts. But somehow only html alerts gets triggered. 

**How to reproduce it (as minimally and precisely as possible)**:
In Rancher cluster try sending email alerts in using text template:
**Environment**
Rancher v2.6.9
* Prometheus Operator version:

  v0.59.1 and also tested for v0.50.0(same issue)
    
    

* Kubernetes version information:

  Client Version: version.Info{Major:"1", Minor:"12", GitVersion:"v1.12.0",
    

* Kubernetes cluster kind:

    insert how you created your cluster: kops, bootkube, etc.

* Manifests:

```
global:
  resolve_timeout: 5m
  smtp_from: reply@xyz
  smtp_smarthost: abc:25
  smtp_require_tls: false
  smtp_auth_password: xxxxxx
  smtp_auth_username: xxxxx
  
route:
  group_by: ['job']
  receiver: email_platform
  routes:
  - receiver: 'null'
    matchers:
      - alertname =~ "InfoInhibitor|Watchdog"
  - receiver: email_platform
    continue: true
receivers:
- name: email_platform
  # override subject line and html content 
  email_configs:
  - to: abc@zxc
    send_resolved: true
    headers:
      subject: '{{ template "email.sm9.subject" . }}'
    html: '{{ define "itsmhtml" }} {{end}}'
    text: '{{ template "email.sm9.text" . }}'
- name: 'null'
templates:
- /etc/alertmanager/*.tmpl
```

* Prometheus Operator Logs:

```
<html>
<body>
<!--StartFragment-->

  |  
-- | --
Fri, Mar 17 2023 12:01:22 pm | level=info ts=2023-03-17T06:31:22.280603822Z caller=main.go:261 msg="Starting Prometheus Operator" version="(version=)"
Fri, Mar 17 2023 12:01:22 pm | level=info ts=2023-03-17T06:31:22.280707533Z caller=main.go:262 build_context="(go=go1.16, user=paulfantom, date=20210817-15:18:20)"
Fri, Mar 17 2023 12:01:22 pm | level=info ts=2023-03-17T06:31:22.287934833Z caller=server.go:54 msg="enabling server side TLS"
Fri, Mar 17 2023 12:01:22 pm | level=info ts=2023-03-17T06:31:22.478835736Z caller=operator.go:449 component=alertmanageroperator msg="connection established" cluster-version=v1.21.5
Fri, Mar 17 2023 12:01:22 pm | level=info ts=2023-03-17T06:31:22.478910696Z caller=operator.go:458 component=alertmanageroperator msg="CRD API endpoints ready"
Fri, Mar 17 2023 12:01:22 pm | level=info ts=2023-03-17T06:31:22.479027523Z caller=operator.go:309 component=thanosoperator msg="connection established" cluster-version=v1.21.5
Fri, Mar 17 2023 12:01:22 pm | level=info ts=2023-03-17T06:31:22.479044889Z caller=operator.go:318 component=thanosoperator msg="CRD API endpoints ready"
Fri, Mar 17 2023 12:01:22 pm | level=info ts=2023-03-17T06:31:22.479180119Z caller=operator.go:421 component=prometheusoperator msg="connection established" cluster-version=v1.21.5
Fri, Mar 17 2023 12:01:22 pm | level=info ts=2023-03-17T06:31:22.479207599Z caller=operator.go:430 component=prometheusoperator msg="CRD API endpoints ready"
Fri, Mar 17 2023 12:01:22 pm | level=info ts=2023-03-17T06:31:22.578523183Z caller=main.go:124 msg="Starting secure server on [::]:8443"
Fri, Mar 17 2023 12:01:24 pm | level=info ts=2023-03-17T06:31:24.079345898Z caller=operator.go:261 component=thanosoperator msg="successfully synced all caches"
Fri, Mar 17 2023 12:01:28 pm | level=info ts=2023-03-17T06:31:28.779706517Z caller=operator.go:351 component=prometheusoperator msg="successfully synced all caches"
Fri, Mar 17 2023 12:01:28 pm | level=info ts=2023-03-17T06:31:28.780037609Z caller=operator.go:1221 component=prometheusoperator key=cattle-monitoring-system/rancher-monitoring-prometheus msg="sync prometheus"
Fri, Mar 17 2023 12:01:30 pm | level=info ts=2023-03-17T06:31:30.079395366Z caller=operator.go:281 component=alertmanageroperator msg="successfully synced all caches"
Fri, Mar 17 2023 12:01:33 pm | level=info ts=2023-03-17T06:31:33.74454625Z caller=operator.go:1306 component=prometheusoperator key=cattle-monitoring-system/rancher-monitoring-prometheus statefulset=prometheus-rancher-monitoring-prometheus shard=0 msg="recreating StatefulSet because the update operation wasn't possible" reason="Forbidden: updates to statefulset spec for fields other than 'replicas', 'template', and 'updateStrategy' are forbidden"
Fri, Mar 17 2023 12:01:33 pm | level=info ts=2023-03-17T06:31:33.779267507Z caller=operator.go:1221 component=prometheusoperator key=cattle-monitoring-system/rancher-monitoring-prometheus msg="sync prometheus"
Fri, Mar 17 2023 12:01:35 pm | level=info ts=2023-03-17T06:31:35.654985761Z caller=operator.go:1306 component=prometheusoperator key=cattle-monitoring-system/rancher-monitoring-prometheus statefulset=prometheus-rancher-monitoring-prometheus shard=0 msg="recreating StatefulSet because the update operation wasn't possible" reason="Forbidden: updates to statefulset spec for fields other than 'replicas', 'template', and 'updateStrategy' are forbidden"
Fri, Mar 17 2023 12:01:35 pm | level=info ts=2023-03-17T06:31:35.664279732Z caller=operator.go:1221 component=prometheusoperator key=cattle-monitoring-system/rancher-monitoring-prometheus msg="sync prometheus"
Fri, Mar 17 2023 12:01:37 pm | level=info ts=2023-03-17T06:31:37.521056505Z caller=operator.go:1221 component=prometheusoperator key=cattle-monitoring-system/rancher-monitoring-prometheus msg="sync prometheus"
Fri, Mar 17 2023 12:01:39 pm | level=info ts=2023-03-17T06:31:39.432489106Z caller=operator.go:1221 component=prometheusoperator key=cattle-monitoring-system/rancher-monitoring-prometheus msg="sync prometheus"
Fri, Mar 17 2023 12:01:41 pm | level=info ts=2023-03-17T06:31:41.73580063Z caller=operator.go:1221 component=prometheusoperator key=cattle-monitoring-system/rancher-monitoring-prometheus msg="sync prometheus"
Fri, Mar 17 2023 12:01:48 pm | level=info ts=2023-03-17T06:31:48.933941554Z caller=operator.go:1221 component=prometheusoperator key=cattle-monitoring-system/rancher-monitoring-prometheus msg="sync prometheus"
Fri, Mar 17 2023 12:01:51 pm | level=info ts=2023-03-17T06:31:51.147879591Z caller=operator.go:1221 component=prometheusoperator key=cattle-monitoring-system/rancher-monitoring-prometheus msg="sync prometheus"
Fri, Mar 17 2023 12:01:53 pm | level=info ts=2023-03-17T06:31:53.636547517Z caller=operator.go:1221 component=prometheusoperator key=cattle-monitoring-system/rancher-monitoring-prometheus msg="sync prometheus"
Fri, Mar 17 2023 12:59:48 pm | level=info ts=2023-03-17T07:29:48.046544388Z caller=operator.go:1221 component=prometheusoperator key=cattle-monitoring-system/rancher-monitoring-prometheus msg="sync prometheus"
Fri, Mar 17 2023 12:59:50 pm | level=info ts=2023-03-17T07:29:50.380127554Z caller=operator.go:1221 component=prometheusoperator key=cattle-monitoring-system/rancher-monitoring-prometheus msg="sync prometheus"
Fri, Mar 17 2023 1:00:35 pm | level=info ts=2023-03-17T07:30:35.737901261Z caller=operator.go:1221 component=prometheusoperator key=cattle-monitoring-system/rancher-monitoring-prometheus msg="sync prometheus"
Fri, Mar 17 2023 1:00:37 pm | level=info ts=2023-03-17T07:30:37.601240396Z caller=operator.go:1221 component=prometheusoperator key=cattle-monitoring-system/rancher-monitoring-prometheus msg="sync prometheus"
Fri, Mar 17 2023 1:00:39 pm | level=info ts=2023-03-17T07:30:39.578351457Z caller=operator.go:1221 component=prometheusoperator key=cattle-monitoring-system/rancher-monitoring-prometheus msg="sync prometheus"
Fri, Mar 17 2023 1:00:41 pm | level=info ts=2023-03-17T07:30:41.534565147Z caller=operator.go:1221 component=prometheusoperator key=cattle-monitoring-system/rancher-monitoring-prometheus msg="sync prometheus"
Fri, Mar 17 2023 1:05:36 pm | level=info ts=2023-03-17T07:35:36.585818531Z caller=operator.go:1221 component=prometheusoperator key=cattle-monitoring-system/rancher-monitoring-prometheus msg="sync prometheus"
Fri, Mar 17 2023 1:05:38 pm | level=info ts=2023-03-17T07:35:38.546573469Z caller=operator.go:1221 component=prometheusoperator key=cattle-monitoring-system/rancher-monitoring-prometheus msg="sync prometheus"
Fri, Mar 17 2023 1:06:31 pm | level=info ts=2023-03-17T07:36:31.870932703Z caller=operator.go:1221 component=prometheusoperator key=cattle-monitoring-system/rancher-monitoring-prometheus msg="sync prometheus"
Fri, Mar 17 2023 1:06:33 pm | level=info ts=2023-03-17T07:36:33.740183947Z caller=operator.go:1221 component=prometheusoperator key=cattle-monitoring-system/rancher-monitoring-prometheus msg="sync prometheus"
Fri, Mar 17 2023 1:06:36 pm | level=info ts=2023-03-17T07:36:36.063363369Z caller=operator.go:1221 component=prometheusoperator key=cattle-monitoring-system/rancher-monitoring-prometheus msg="sync prometheus"
Fri, Mar 17 2023 1:06:37 pm | level=info ts=2023-03-17T07:36:37.702629782Z caller=operator.go:1221 component=prometheusoperator key=cattle-monitoring-system/rancher-monitoring-prometheus msg="sync prometheus"
Fri, Mar 17 2023 1:07:45 pm | level=info ts=2023-03-17T07:37:45.009205264Z caller=operator.go:1221 component=prometheusoperator key=cattle-monitoring-system/rancher-monitoring-prometheus msg="sync prometheus"
Fri, Mar 17 2023 1:07:47 pm | level=info ts=2023-03-17T07:37:47.137408275Z caller=operator.go:1221 component=prometheusoperator key=cattle-monitoring-system/rancher-monitoring-prometheus msg="sync prometheus"
Fri, Mar 17 2023 1:07:48 pm | level=info ts=2023-03-17T07:37:48.79059118Z caller=operator.go:1221 component=prometheusoperator key=cattle-monitoring-system/rancher-monitoring-prometheus msg="sync prometheus"
Fri, Mar 17 2023 1:11:37 pm | level=info ts=2023-03-17T07:41:37.920279978Z caller=operator.go:1221 component=prometheusoperator key=cattle-monitoring-system/rancher-monitoring-prometheus msg="sync prometheus"
Fri, Mar 17 2023 1:13:33 pm | level=info ts=2023-03-17T07:43:33.469644995Z caller=operator.go:1221 component=prometheusoperator key=cattle-monitoring-system/rancher-monitoring-prometheus msg="sync prometheus"
Fri, Mar 17 2023 1:13:35 pm | level=info ts=2023-03-17T07:43:35.278419487Z caller=operator.go:1221 component=prometheusoperator key=cattle-monitoring-system/rancher-monitoring-prometheus msg="sync prometheus"

<!--EndFragment-->
</body>
</html>
```

**Anything else we need to know?**:
Initially deployed Rancher provided V2 monitoring consisting of alert manager and tried to sent custom text email alerts but we were getting alerts in html template only, so we passed blank html template but to suppress html template then then we were just getting blank email alerts. So we installed Rancher monitoring without alertmanager and  deployed-integrated out of the box alertmanager but still the same issue.

### PR DESCRIPTION
**Pull request by simonpasquier**
## Description

Closes #5421

## Type of change

_What type of changes does your code introduce to the Prometheus operator? Put an `x` in the box that apply._

- [ ] `CHANGE` (fix or feature that would cause existing functionality to not work as expected)
- [ ] `FEATURE` (non-breaking change which adds functionality)
- [X] `BUGFIX` (non-breaking change which fixes an issue)
- [ ] `ENHANCEMENT` (non-breaking change which improves existing functionality)
- [ ] `NONE` (if none of the other choices apply. Example, tooling, build system, CI, docs, etc.)

## Changelog entry

_Please put a one-line changelog entry below. This will be copied to the changelog file during the release process._

```release-note
fix: support text-only email configs
```

### CODE REVIEW

**Comment by simonpasquier** [comment]
I've tested locally: when setting `html: ""` in the Alertmanager configuration secret, I receive a text-only email.

File: pkg/apis/monitoring/v1alpha1/alertmanager_config_types.go [hunk 1]
```diff
@@ -699,10 +699,10 @@ type EmailConfig struct {
 	Headers []KeyValue `json:"headers,omitempty"`
 	// The HTML body of the email notification.
 	// +optional
-	HTML string `json:"html,omitempty"`
+	HTML *string `json:"html,omitempty"`
```
Comment (nicolastakashi) [inline pkg/apis/monitoring/v1alpha1/alertmanager_config_types.go:702]: Do still wanna make improvements and bug fixed on alpha apis?

Comment (simonpasquier) [reply pkg/apis/monitoring/v1alpha1/alertmanager_config_types.go:702]: for now, v1alpha1 is still the default version being installed. I started #5398 a while ago to move forward with v1beta1 but no time to complete it for now...

Comment (nicolastakashi) [reply pkg/apis/monitoring/v1alpha1/alertmanager_config_types.go:702]: Got it, thx

**Review by ArthurSens — APPROVED** [review]
Nice job tracking this down all the way from an issue in prometheus/alertmanager :)

**Review by philipgough — APPROVED** [review]

**Pull request merged**